### PR TITLE
Support RpcHttpBodyOnly and RpcHttpTriggerMetadataRemoved

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
This is the implementation of the issue Support "RpcHttpBodyOnly"capability #319
https://github.com/Azure/azure-functions-java-worker/issues/319

This PR includes: 
* Support RpcHttpBodyOnly capability
* Support RpcHttpTriggerMetadataRemoved
* Mocking gRPC using jMockit
* Resolve the test failure of binding port for passing all test

This is the joint effort with @amamounelsayed 

### RpcHttpTriggerMetadataRemoved
If you don't have RpcHttpTriggerMetadataRemoved, you will find the duplicate data between inputData.0.data_ and triggerMetadata.mapData.$request and "req" These are the identical and requires to send the same json three times. 

image

Then if you specify RpcHttpTriggerMetadataRemoved (and RpcHttpBodyOnly) you will find the triggerMetadata duplication is removed.

image

However, for keeping the same experience for the Azure Functions invocation, we need to copy the "req" and "$request" from inputData.0.data

That is why I implement the method.  This is the same implementation as node. 

## NOTE
The test package structure is not identical as the main package. I recommend to make it the same structure. Then we can use package private for testing methods. Currently I had to make it public just for testability of unit testing. (If you don't like it, I'd like to make it separate class and test it).

### CODE REVIEW

File: src/main/java/com/microsoft/azure/functions/worker/binding/RpcHttpRequestDataSource.java [hunk 1]
```diff
@@ -21,7 +21,8 @@
 	public RpcHttpRequestDataSource(String name, RpcHttp value) {
 		super(name, null, HTTP_DATA_OPERATIONS);
 		this.httpPayload = value;
-		this.bodyDataSource = BindingDataStore.rpcSourceFromTypedData(null, this.httpPayload.getBody());
+//		this.bodyDataSource = BindingDataStore.rpcSourceFromTypedData(null, this.httpPayload.getBody());
```
Comment (pragnagopa) [inline src/main/java/com/microsoft/azure/functions/worker/binding/RpcHttpRequestDataSource.java:24]: nit: remove commented code.

File: src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java [hunk 2]
```diff
@@ -59,6 +57,29 @@ public void loadMethod(FunctionMethodDescriptor descriptor, Map<String, BindingI
...
+				name = e.getName();
+				break;
+			}
+		}
+
+		Map<String, TypedData> triggerMetadata = new HashMap(request.getTriggerMetadataMap());
+		if (!name.isEmpty() && !triggerMetadata.containsKey(name)) {
+			triggerMetadata.put(name, dataWithHttp);
+		}
+		String requestKey = "$request";
```
Comment (pragnagopa) [inline src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java:76]: nit: move to constants file

**Comment by mathewc** [comment]
In my PR https://github.com/Azure/azure-functions-host/pull/5897, I'm no longer sending ANY duplicate htttp request data via BindingData, since the request details are already sent as part of inputs. With your changes, will things work with none of these request details being in the binding data?

**Comment by TsuyoshiUshio** [comment]
@mathewc Yes. It works. My code takes value from inputData to create triggerMetadate like req, $request.  I need research how Java Azure Functions side uses these triggerMetadata, however, if we can create the triggerMetadata from inputData, it must be fine. I'll do the research for the usage of the triggerMetadata and will share in here.

File: src/main/java/com/microsoft/azure/functions/worker/binding/RpcHttpRequestDataSource.java (hunk 1 above)
Comment (TsuyoshiUshio) [reply src/main/java/com/microsoft/azure/functions/worker/binding/RpcHttpRequestDataSource.java]: Done

File: src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java (hunk 2 above)
Comment (TsuyoshiUshio) [reply src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java]: Done

**Comment by TsuyoshiUshio** [comment]
@pragnagopa I've done the fix. Also I add one more test scenario (if the request is non-http request, no trigger metadata is modified). @amamounelsayed if you need to discuss more, please leave your message. If @amamounelsayed  has no comment, let's get this pr merged!

**Comment by amamounelsayed** [comment]
Thank you @TsuyoshiUshio I will review it and merge sir.

**Comment by amamounelsayed** [comment]
@TsuyoshiUshio I found an issue with RpcHttpBodyOnly  logic, that is initially I added. I fixed it. I am writing tests for it.

**Pull request merged**